### ISSUE TITLE
Could this metadata be incorporated please?

### ISSUE THREAD

**Issue by bluefiberbread** [ccloli/E-Hentai-Downloader#148]
https://github.com/ccloli/E-Hentai-Downloader/pull/140/commits/af11667d7b8fa2b92fca3b05ac074e6fd7ecdfa6

**Comment by ccloli** [comment]
Hi, I've seen the issue but forgot to reply, sorry for the late reply.

However I didn't see the benefit of JSON format, the readability is not as clear as a plain text file for common user, unless you're using some automation tools or it's formatted. Also the JSON metadata provided by `x/gallery-metadata` is not fully compatible with E-Hentai official JSON API schema. 

```js
{
  "gid": 592178,
  "token": "41cc263dc7",
  "archiver_key": "435566--04913597254dc4fb1e655bdda9a8be4fb9878df1",
  "title": "(Shota Scratch 5) [Studio Zealot (Various)] Bokutachi! Shotappuru!! (Boku no Pico)",
  "title_jpn": "(ショタスクラッチ5) [Studio Zealot (よろず)] ぼくたち!しょたっぷる!! (ぼくのぴこ)",
  "category": "Doujinshi",
  "thumb": "https:\/\/ehgt.org\/4c\/6a\/4c6ad39fffcdefcb2cd35218a95395af2e5ad74d-1854978-2118-3000-jpg_l.jpg",
  "uploader": "tooecchi",
  "posted": "1368418878",
  "filecount": "63",
  "filesize": 75630519,
  "expunged": false,
  "rating": "4.54",
  "torrentcount": "1",
  "torrents": [
    {
      "hash": "2a4641feba9943b0e028927879ff6567e74bf0ae",
      "added": "1551314399",
      "name": "(Shota Scratch 5) [Studio Zealot (Various)] Bokutachi! Shotappuru!! (Boku no Pico)",
      "tsize": "15576",
      "fsize": "75637278"
    }
  ],
  "tags": [
    "parody:boku no pico",
    "group:studio zealot",
    "artist:aoi madoka",
    "artist:azuma kyouto",
    "artist:fujinomiya yuu",
    "artist:mikami hokuto",
    "artist:miyakawa hajime",
    "artist:mokkouyou bond",
    "artist:murasaki nyaa",
    "artist:nemunemu",
    "artist:po-ju",
    "artist:rustle",
    "artist:tanuma yuuichirou",
    "artist:tower",
    "male:anal",
    "male:bloomers",
    "male:catboy",
    "male:crossdressing",
    "male:frottage",
    "male:josou seme",
    "male:school swimsuit",
    "male:schoolgirl uniform",
    "male:shotacon",
    "male:tomgirl",
    "male:yaoi"
  ]
}
```

So at this time I'm not willing to merge the PR for some uncommon features, and it may reduce maintainability (thought 1.x is already a mess).

**Comment by pencilbook** [comment]
x/gallery-metadata's JSON provides more information and there are tools for managing and reading comics/doujins (HappyPandaX and LANraragi) that can automatically fetch information from x/gallery-metadata's JSON.
JSON is also a format of plain text btw.

I wouldn't say it's uncommon to collect metadata, especially when E-Hentai's owner almost killed the entire website a couple of months ago. This would be an excellent feature but it's up to you to implement it. I don't think anybody would be against it.

**Comment by pencilbook** [comment]
needs to be added, please don't ignore this amazing request.

**Comment by pencilbook** [comment]
Why do you insist on fragmenting metadata support

**Comment by Asinin3** [comment]
Yeah, this is really the only thing stopping me from using E-H Downloader. Meta-data is a huge part of archiving, and when E-H could be taken down at any moment, keeping metadata for galleries is quite important. As Eze has been depreciated and often causes browsers like FireFox to hang I'd love to switch to using E-H Downloader.

**Comment by pencilbook** [comment]
[#140](https://github.com/ccloli/E-Hentai-Downloader/pull/140) worked fine until last month [because exhentai changed the header](https://github.com/ccloli/E-Hentai-Downloader/issues/153#issuecomment-593392614)

Because there were no issues with [#140](https://github.com/ccloli/E-Hentai-Downloader/pull/140), and because we've been using it for years with eze downloader which became deprecated a while ago (which is why we're using EHDL), could this be added to the script now? Lanraragi doesn't support EHDL's gallery info because it's not better than this [#140](https://github.com/ccloli/E-Hentai-Downloader/pull/140).

**Comment by pencilbook** [comment]
@LoLsAlEd merged EHDL with the official metadata userscript a few days ago. It works for me with Firefox and Violentmonkey but I cannot guarantee it'll work for you.

link to the userscript: https://gist.github.com/LoLsAlEd/642a67a3e67cef0e459c44aaa4e6bf88

**Comment by Asinin3** [comment]
> @LoLsAlEd merged EHDL with the official metadata userscript a few days ago. It works for me with Firefox and Violentmonkey but I cannot guarantee it'll work for you.
> 
> link to the userscript: https://gist.github.com/LoLsAlEd/642a67a3e67cef0e459c44aaa4e6bf88

Shame that the script broke and it cant download original files anymore... @LoLsAlEd Is it going to be fixed?

**Comment by ccloli** [comment]
See https://github.com/ccloli/E-Hentai-Downloader/pull/140#issuecomment-2094686556

Or tl;dr, https://github.com/ccloli/E-Hentai-Downloader/tree/dnsev-h_x-info
